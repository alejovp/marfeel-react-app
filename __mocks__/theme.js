export const themeMock = {
    tipography: {
        primaryFontColor: 'black',
        fontFamily: 'Helvetica, sans-serif'
    },
    header: {
        padding: 15
    },
    sectionMenu: {
        color: 'orange',
        textSize: 1.1 //em
    },
    background: {
        color: 'gray'
    },
    burgerButton: {
        color: 'red'
    }
};
