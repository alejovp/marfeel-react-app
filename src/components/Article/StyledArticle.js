import styled from 'styled-components';
import { device } from '../../styles/device';


export const StyledArticle = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
    padding: 20px;
    flex-wrap: wrap;

    @media ${device.tablet} {
        flex-wrap: nowrap;
        align-items: unset;
    }
`;

StyledArticle.displayName = 'StyledArticle';
