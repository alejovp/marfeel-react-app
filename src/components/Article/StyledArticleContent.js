import styled from 'styled-components';

export const StyledArticleContent = styled.div`
    display: flex;
    flex-direction: column;
`;

StyledArticleContent.displayName = 'StyledArticleContent';
