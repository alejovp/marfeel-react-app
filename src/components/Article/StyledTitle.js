import styled from 'styled-components';

export const StyledTitle = styled.h3`
    margin-top: 0;
`;

StyledTitle.displayName = 'StyledTitle';
