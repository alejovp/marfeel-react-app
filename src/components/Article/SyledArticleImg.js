import styled from 'styled-components';
import { device } from '../../styles/device';

export const StyledArticleImg = styled.img`
    width: 200px;
    height: 150px;
    object-fit: cover;
    margin-bottom: 20px;

    @media ${device.mobileL} { 
        width: 350px;
        height: 250px;
    }

    @media ${device.tablet} {
        margin-bottom: 0;
        margin-right: 20px;
    }
`;

StyledArticleImg.displayName = 'StyledArticleImg';
