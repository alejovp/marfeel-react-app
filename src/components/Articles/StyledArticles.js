import styled from 'styled-components';
import { device } from '../../styles/device';


export const StyledArticles = styled.section`
    padding: 10px;

    @media ${device.tablet} {
        padding: 50px;
    }
`;

StyledArticles.displayName = 'StyledArticles';
