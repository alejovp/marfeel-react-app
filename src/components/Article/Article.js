import React from 'react';
import PropTypes from 'prop-types';
import { StyledArticle } from './StyledArticle';
import { StyledArticleContent } from './StyledArticleContent';
import { StyledTitle } from './StyledTitle';
import { StyledArticleImg } from './SyledArticleImg';


export const Article = ({ img, title, description}) => {

    if (!title || !description) {
        return null;
    }

    const renderImage = () => {
        if (!img) {
            return null;
        }
        return <StyledArticleImg src={img} />;
    };

    return (
        <StyledArticle>
            {renderImage()}
            <StyledArticleContent>
                <StyledTitle>
                    {title}
                </StyledTitle>
                <div>
                    <p>{description}</p>
                </div>
            </StyledArticleContent>
        </StyledArticle>
    );
};

Article.propTypes = {
    img: PropTypes.string,
    title: PropTypes.string,
    description: PropTypes.string
};
