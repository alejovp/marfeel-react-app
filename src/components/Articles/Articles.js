import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Article } from '../../components/Article';
import { StyledArticles } from './StyledArticles';
import { Divider } from '../Divider/Divider';


export const Articles = ({ articles = [] }) => {

    const renderArticles = () => articles.map(article => (
        <Fragment key={article.id}>
            <Article img={article.img}
                     title={article.title}
                     description={article.description} />
            <Divider />
        </Fragment>
    ));

    return (
        <StyledArticles>
            {renderArticles()}
        </StyledArticles>
    );
};

Articles.propTypes = {
    articles: PropTypes.array
};
