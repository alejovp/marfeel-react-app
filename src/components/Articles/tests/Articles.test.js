import React from 'react';
import { Articles } from '../Articles';
import theme from '../../../../__mocks__/theme.json';
import { shallowWithTheme } from '../../../../scripts/jest/setup';
import articlesMock from '../../../../__mocks__/articles.json';
import { StyledArticles } from '../StyledArticles';
import { Article } from '../../Article/Article';


describe('<Articles />', () => {

    describe('shape', () => {
        const renderedComponent = shallowWithTheme(
            <Articles articles={articlesMock.items} />, theme);

        it('will render correctly', () => {
            expect(renderedComponent).toMatchSnapshot();
        });

        it('will render a StyledArticles component', () => {
            expect(renderedComponent.find(StyledArticles)).toHaveLength(1);
        });

        it('will render as many Article comps as items in articlesMock', () => {
            expect(renderedComponent.find(Article)).toHaveLength(articlesMock.items.length);
        });

    });

});
