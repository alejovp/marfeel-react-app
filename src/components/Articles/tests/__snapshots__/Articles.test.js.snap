// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Articles /> shape will render correctly 1`] = `
<StyledArticles>
  <Article
    description="Excepteur laborum quis anim dolor dolore proident culpa mollit ea aliquip elit culpa. Non voluptate qui Lorem laborum ad ex ipsum excepteur incididunt esse. Aliquip sunt in est voluptate cillum pariatur. Magna tempor labore esse quis dolor esse voluptate. Labore anim aliquip sint ullamco consectetur ad exercitation eu proident."
    img="https://burst.shopifycdn.com/photos/business-flatlay-in-india.jpg?width=373&format=pjpg&exif=0&iptc=0"
    title="Aute sunt nulla nostrud exercitation amet voluptate Lorem cillum incididunt proident."
  />
  <Divider />
  <Article
    description="Pariatur mollit quis occaecat in qui aute aute. Qui aute adipisicing et ea sit veniam ut officia eu pariatur tempor deserunt. Commodo consectetur ipsum Lorem magna quis labore ullamco est. Enim est enim fugiat aliquip veniam non. Sunt sunt proident dolore qui ipsum veniam incididunt exercitation adipisicing ut ex."
    img="https://burst.shopifycdn.com/photos/woman-edits-images-on-her-laptop.jpg?width=373&format=pjpg&exif=0&iptc=0"
    title="Sint consectetur et aute qui qui reprehenderit incididunt incididunt sunt."
  />
  <Divider />
  <Article
    description="Amet aliqua eiusmod Lorem minim consectetur non esse sint non commodo veniam cillum. Anim labore labore excepteur dolor cupidatat incididunt dolor quis magna nulla nostrud est. Reprehenderit eiusmod dolor officia deserunt cupidatat non ipsum. Anim non irure amet exercitation aute duis laboris enim dolor incididunt irure aute. Qui aliquip incididunt reprehenderit occaecat aliqua adipisicing sint ex nostrud ad mollit."
    img="https://burst.shopifycdn.com/photos/a-blue-point-himalayan-sits-at-a-desk.jpg?width=373&format=pjpg&exif=0&iptc=0"
    title="Voluptate mollit elit ipsum eiusmod voluptate dolor pariatur tempor ex."
  />
  <Divider />
  <Article
    description="Commodo ad ut proident nisi nisi ut. Magna nostrud officia ipsum magna quis nisi sunt mollit do veniam ex. Ullamco velit consectetur nisi non mollit eiusmod ullamco. Eu duis in ipsum aliquip fugiat elit consectetur ad."
    img="https://burst.shopifycdn.com/photos/design-plans-color-swatches.jpg?width=373&format=pjpg&exif=0&iptc=0"
    title="Do laborum dolore qui adipisicing occaecat aute culpa amet proident."
  />
  <Divider />
  <Article
    description="Excepteur laborum quis anim dolor dolore proident culpa mollit ea aliquip elit culpa. Non voluptate qui Lorem laborum ad ex ipsum excepteur incididunt esse. Aliquip sunt in est voluptate cillum pariatur. Magna tempor labore esse quis dolor esse voluptate. Labore anim aliquip sint ullamco consectetur ad exercitation eu proident."
    img="https://burst.shopifycdn.com/photos/desk-setup-with-motivational-poster.jpg?width=373&format=pjpg&exif=0&iptc=0"
    title="Aute sunt nulla nostrud exercitation amet voluptate Lorem cillum incididunt proident."
  />
  <Divider />
  <Article
    description="Excepteur laborum quis anim dolor dolore proident culpa mollit ea aliquip elit culpa. Non voluptate qui Lorem laborum ad ex ipsum excepteur incididunt esse. Aliquip sunt in est voluptate cillum pariatur. Magna tempor labore esse quis dolor esse voluptate. Labore anim aliquip sint ullamco consectetur ad exercitation eu proident."
    img=""
    title="Aute sunt nulla nostrud exercitation amet voluptate Lorem cillum incididunt proident."
  />
  <Divider />
  <Article
    description="Pariatur mollit quis occaecat in qui aute aute. Qui aute adipisicing et ea sit veniam ut officia eu pariatur tempor deserunt. Commodo consectetur ipsum Lorem magna quis labore ullamco est. Enim est enim fugiat aliquip veniam non. Sunt sunt proident dolore qui ipsum veniam incididunt exercitation adipisicing ut ex."
    img="https://burst.shopifycdn.com/photos/woman-edits-images-on-her-laptop.jpg?width=373&format=pjpg&exif=0&iptc=0"
    title="Sint consectetur et aute qui qui reprehenderit incididunt incididunt sunt."
  />
  <Divider />
  <Article
    description="Amet aliqua eiusmod Lorem minim consectetur non esse sint non commodo veniam cillum. Anim labore labore excepteur dolor cupidatat incididunt dolor quis magna nulla nostrud est. Reprehenderit eiusmod dolor officia deserunt cupidatat non ipsum. Anim non irure amet exercitation aute duis laboris enim dolor incididunt irure aute. Qui aliquip incididunt reprehenderit occaecat aliqua adipisicing sint ex nostrud ad mollit."
    img="https://burst.shopifycdn.com/photos/a-blue-point-himalayan-sits-at-a-desk.jpg?width=373&format=pjpg&exif=0&iptc=0"
    title="Voluptate mollit elit ipsum eiusmod voluptate dolor pariatur tempor ex."
  />
  <Divider />
  <Article
    description="Commodo ad ut proident nisi nisi ut. Magna nostrud officia ipsum magna quis nisi sunt mollit do veniam ex. Ullamco velit consectetur nisi non mollit eiusmod ullamco. Eu duis in ipsum aliquip fugiat elit consectetur ad."
    img="https://burst.shopifycdn.com/photos/design-plans-color-swatches.jpg?width=373&format=pjpg&exif=0&iptc=0"
    title="Do laborum dolore qui adipisicing occaecat aute culpa amet proident."
  />
  <Divider />
  <Article
    description="Excepteur laborum quis anim dolor dolore proident culpa mollit ea aliquip elit culpa. Non voluptate qui Lorem laborum ad ex ipsum excepteur incididunt esse. Aliquip sunt in est voluptate cillum pariatur. Magna tempor labore esse quis dolor esse voluptate. Labore anim aliquip sint ullamco consectetur ad exercitation eu proident."
    img="https://burst.shopifycdn.com/photos/desk-setup-with-motivational-poster.jpg?width=373&format=pjpg&exif=0&iptc=0"
    title="Aute sunt nulla nostrud exercitation amet voluptate Lorem cillum incididunt proident."
  />
  <Divider />
  <Article
    description="Excepteur laborum quis anim dolor dolore proident culpa mollit ea aliquip elit culpa. Non voluptate qui Lorem laborum ad ex ipsum excepteur incididunt esse. Aliquip sunt in est voluptate cillum pariatur. Magna tempor labore esse quis dolor esse voluptate. Labore anim aliquip sint ullamco consectetur ad exercitation eu proident."
    img="https://burst.shopifycdn.com/photos/business-flatlay-in-india.jpg?width=373&format=pjpg&exif=0&iptc=0"
    title="Aute sunt nulla nostrud exercitation amet voluptate Lorem cillum incididunt proident."
  />
  <Divider />
  <Article
    description="Pariatur mollit quis occaecat in qui aute aute. Qui aute adipisicing et ea sit veniam ut officia eu pariatur tempor deserunt. Commodo consectetur ipsum Lorem magna quis labore ullamco est. Enim est enim fugiat aliquip veniam non. Sunt sunt proident dolore qui ipsum veniam incididunt exercitation adipisicing ut ex."
    img="https://burst.shopifycdn.com/photos/woman-edits-images-on-her-laptop.jpg?width=373&format=pjpg&exif=0&iptc=0"
    title="Sint consectetur et aute qui qui reprehenderit incididunt incididunt sunt."
  />
  <Divider />
  <Article
    description="Amet aliqua eiusmod Lorem minim consectetur non esse sint non commodo veniam cillum. Anim labore labore excepteur dolor cupidatat incididunt dolor quis magna nulla nostrud est. Reprehenderit eiusmod dolor officia deserunt cupidatat non ipsum. Anim non irure amet exercitation aute duis laboris enim dolor incididunt irure aute. Qui aliquip incididunt reprehenderit occaecat aliqua adipisicing sint ex nostrud ad mollit."
    img="https://burst.shopifycdn.com/photos/a-blue-point-himalayan-sits-at-a-desk.jpg?width=373&format=pjpg&exif=0&iptc=0"
    title="Voluptate mollit elit ipsum eiusmod voluptate dolor pariatur tempor ex."
  />
  <Divider />
  <Article
    description="Commodo ad ut proident nisi nisi ut. Magna nostrud officia ipsum magna quis nisi sunt mollit do veniam ex. Ullamco velit consectetur nisi non mollit eiusmod ullamco. Eu duis in ipsum aliquip fugiat elit consectetur ad."
    img="https://burst.shopifycdn.com/photos/design-plans-color-swatches.jpg?width=373&format=pjpg&exif=0&iptc=0"
    title="Do laborum dolore qui adipisicing occaecat aute culpa amet proident."
  />
  <Divider />
  <Article
    description="Excepteur laborum quis anim dolor dolore proident culpa mollit ea aliquip elit culpa. Non voluptate qui Lorem laborum ad ex ipsum excepteur incididunt esse. Aliquip sunt in est voluptate cillum pariatur. Magna tempor labore esse quis dolor esse voluptate. Labore anim aliquip sint ullamco consectetur ad exercitation eu proident."
    img="https://burst.shopifycdn.com/photos/desk-setup-with-motivational-poster.jpg?width=373&format=pjpg&exif=0&iptc=0"
    title="Aute sunt nulla nostrud exercitation amet voluptate Lorem cillum incididunt proident."
  />
  <Divider />
</StyledArticles>
`;
